src/main/java/graphql/execution/ValuesResolver.java

```diff
@@ -9,11 +9,13 @@
 
 public class ValuesResolver {
 
-
     public Map<String, Object> getVariableValues(GraphQLSchema schema, List<VariableDefinition> variableDefinitions, Map<String, Object> inputs) {
         Map<String, Object> result = new LinkedHashMap<String, Object>();
         for (VariableDefinition variableDefinition : variableDefinitions) {
-            result.put(variableDefinition.getName(), getVariableValue(schema, variableDefinition, inputs.get(variableDefinition.getName())));
+            Object variableValue = getVariableValue(schema, variableDefinition, inputs.get(variableDefinition.getName()));
+            if (variableValue != null) {
+                result.put(variableDefinition.getName(), variableValue);
+            }
```

Comment:
Ah okay, thank you. That should have been obvious, actually. Makes sense.

But, so then what about people who may be relying on the behavior that's currently in place (my compatibility question)? My feeling is that someone might want the current behavior while someone else might want this behavior. Is there some reason this behavior is more "correct"?